### PR TITLE
added section MXD tutorial 1 data exchange

### PR DESCRIPTION
**Pull request by jkbquabeck**
## Description

This PR publishes the first MXD tutorial on initiating data exchange within the MXD. This adds one tutorial document and one image. This was requested in issue #315 (https://github.com/eclipse-tractusx/eclipse-tractusx.github.io/issues/315). The MXD tutorials are to be published in different repositories, starting with tutorial_resources.

## Pre-review checks

- [ ] DEPENDENCIES are up-to-date. [Dash license tool](https://github.com/eclipse/dash-licenses). Committers can open IP issues for restricted libs.
- [x] [Copyright and license header](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-02) are present on all affected files

### CODE REVIEW

File: mxd/3.1_MXD_tutorial1_dataexchange.md
Comment (paullatzelsperger) [inline mxd/3.1_MXD_tutorial1_dataexchange.md:3]:
```suggestion
As described in the introduction, Bob (Provider) wants to exchange data with Alice (Consumer).
```

File: mxd/3.1_MXD_tutorial1_dataexchange.md
Comment (paullatzelsperger) [inline mxd/3.1_MXD_tutorial1_dataexchange.md:5]: spurious. water is wet.

File: mxd/3.1_MXD_tutorial1_dataexchange.md
Comment (paullatzelsperger) [inline mxd/3.1_MXD_tutorial1_dataexchange.md:9]: that sounds very weird. 

```suggestion
In this step we will focus on inserting data into our provider connector (Alice) using
```

File: mxd/3.1_MXD_tutorial1_dataexchange.md
Comment (paullatzelsperger) [inline mxd/3.1_MXD_tutorial1_dataexchange.md:16]: this table is spurious, and it confuses more than it contributes, as the data shown is also contained in the JSON example further down.

File: mxd/3.1_MXD_tutorial1_dataexchange.md
Comment (paullatzelsperger) [inline mxd/3.1_MXD_tutorial1_dataexchange.md:21]: please do this everywhere:
```suggestion
Action (Bob): Create this asset using the following `curl` command:
```

File: mxd/3.1_MXD_tutorial1_dataexchange.md
Comment (paullatzelsperger) [inline mxd/3.1_MXD_tutorial1_dataexchange.md:44]: that is incorrect. Bob tells Alice nothing.

Comment (paullatzelsperger) [inline mxd/3.1_MXD_tutorial1_dataexchange.md:44]:
```suggestion
Bob tells Alice, that he created an asset, and she should now be able to request it. In the next step, Alice requests a contract offer catalog. In the catalog, all contract offers for Alice are listed.
```

File: mxd/3.1_MXD_tutorial1_dataexchange.md
Comment (paullatzelsperger) [inline mxd/3.1_MXD_tutorial1_dataexchange.md:69]: It's not a childrens' book.
```suggestion
Let´s see if Alice can see the Asset (ID:3).
```

File: mxd/3.1_MXD_tutorial1_dataexchange.md
Comment (paullatzelsperger) [inline mxd/3.1_MXD_tutorial1_dataexchange.md:119]: As it is written, this is incorrect. Contracts are not _defined_, they are _negotiated_. However, if you are referring to `contractDefinitions`, then the sentence must be rewritten to:
```suggestion
**This is the first lesson for this tutorial: A contract definition must be created on the data provider side. It must always contain an asset, an access policy and a contract policy.**
```
You could also link to the relevant [documentation](https://github.com/eclipse-edc/docs/blob/main/developer/handbook.md)

File: mxd/3.1_MXD_tutorial1_dataexchange.md
Comment (paullatzelsperger) [inline mxd/3.1_MXD_tutorial1_dataexchange.md:135]: it is important to understand, and illustrate, that access policy and contract policy are two **distinct** policies, that serve separate purposes. it is bad practice to use the same policy for both!!

File: mxd/3.1_MXD_tutorial1_dataexchange.md
Comment (paullatzelsperger) [inline mxd/3.1_MXD_tutorial1_dataexchange.md:171]:
```suggestion
Congratulations on yor first successful data exchange in your own data space!
```

**Review by paullatzelsperger — CHANGES_REQUESTED** [review]
careful with diagrams: please use the exact terms as they appear in the EDC, i.e. `Asset` rather than `Data Asset`.
Please do not just check in PNGs, because they cannot be altered, they should be generated with PlantUML and the source code should be checked in. You should **not** check in formats, that can only be opened with proprietary tools like Visio!

**Review by FaGru3n — CHANGES_REQUESTED** [review]
Hi @jkbquabeck thanks for your contribution.

For content within the tutorial i guess i´m not the perfect one to validate.

But also want to state out to store diagramms in a format like @paullatzelsperger already mentioned.

FYI referenced PR https://github.com/eclipse-tractusx/eclipse-tractusx.github.io/pull/401

and some help for creating either mermaid or puml files.
## Technical requirements

- [PlantUML](https://plantuml.com/en/) or [Mermaid](https://mermaid.js.org/) must be used to create the static files.
- Live Mermaid Editor: [Mermaid Live Editor](https://mermaid.live/edit)
- Live PlantUML Editor: [PlantUML Live Editor](https://www.planttext.com/)

File: mxd/3.1_MXD_tutorial1_dataexchange.md
Comment (jkbquabeck) [reply mxd/3.1_MXD_tutorial1_dataexchange.md:3]: Why dont we call the participants Data Provider and Data Consumer? These are the terms stated in the new regulatory framework.

File: mxd/3.1_MXD_tutorial1_dataexchange.md
Comment (jkbquabeck) [reply mxd/3.1_MXD_tutorial1_dataexchange.md:135]: @hemantxpatel

**Comment by jkbquabeck** [comment]
@hemantxpatel I updated the contents according to Paul's change requests. Would you take a look at the contract policy he is referring to and provide a fitting one, distinct from policy with ID:3?

File: mxd/3.1_MXD_tutorial1_dataexchange.md
Comment (hemantxpatel) [reply mxd/3.1_MXD_tutorial1_dataexchange.md:131]: @jkbquabeck  create a contract policy just above this step.
```
```shell
curl --location 'http://localhost/bob/management/v2/policydefinitions' \
--header 'Content-Type: application/json' \
--header 'X-Api-Key: password' \
--data-raw '{
  "@context": {
    "odrl": "http://www.w3.org/ns/odrl/2/"
  },
  "@type": "PolicyDefinitionRequestDto",
  "@id": "32",
  "policy": {
    "@type": "Policy"
  }
}' 
```
```

File: mxd/3.1_MXD_tutorial1_dataexchange.md
Comment (hemantxpatel) [inline mxd/3.1_MXD_tutorial1_dataexchange.md:132]: Update this line as `"contractPolicyId": "32",`

I think we can update access policy id as 31, so that we will have
`assetId: 3`
`accessPolicyId: 31`
`contractpolicyId: 32`
`contractDefinitionId: 3`

We have used `access policy id: 4` in next tutorial so, we will have to update those. We can avoid if we follow id convention as above.

File: mxd/3.1_MXD_tutorial1_dataexchange.md
Comment (hemantxpatel) [inline mxd/3.1_MXD_tutorial1_dataexchange.md:77]: We can update id as `31` so, access policy id will be 31, and contract policy id will be 32.

**Comment by jkbquabeck** [comment]
@hemantxpatel  Changed curl commands as requested. @paullatzelsperger there is one comment still open.
@stephanbcbauer once this PR is approved you can use the updated contents in the github.io repo feat/tutorials branch.
If approval is granted, do not merge this PR but close it, to avoid redundancy.

File: mxd/3.1_MXD_tutorial1_dataexchange.md
Comment (hemantxpatel) [inline mxd/3.1_MXD_tutorial1_dataexchange.md:137]: Some invalid data,  in request body. For safer side, don't copy it manually, copy via dedicated copy button on the right top of the code viewer.
```shell
curl --location 'http://localhost/bob/management/v2/policydefinitions' \
--header 'Content-Type: application/json' \
--header 'X-Api-Key: password' \
--data-raw '{
  "@context": {
    "odrl": "http://www.w3.org/ns/odrl/2/"
  },
  "@type": "PolicyDefinitionRequestDto",
  "@id": "32",
  "policy": {
    "@type": "Policy"
  }
}'
```

**Comment by stephanbcbauer** [comment]
> @hemantxpatel Changed curl commands as requested. @paullatzelsperger there is one comment still open. @stephanbcbauer once this PR is approved you can use the updated contents in the github.io repo feat/tutorials branch. If approval is granted, do not merge this PR but close it, to avoid redundancy.

@jkbquabeck ok thx. But until now its not approved :)

**Comment by hemantxpatel** [comment]
@jkbquabeck Could you please incorporate review comments. Can we get this PR merged since it's pending since long.

**Pull request closed without merging**
